### DISCUSSION TITLE
dyadic products

### DISCUSSION THREAD

**Discussion by jeanmichelscherer** [thelfer/tfel#48] [Q&A] [answered]
Hi Thomas,

I would like to use a dyadic product which for two (non-symmetric) second-order tensors ```A_ij``` and ```B_kl``` returns a fourth-order tensor ```C_ijkl = A_il*B_jk```. Another useful product would be the one returning ```C_ijkl = A_ik*B_jl```, although I can for now use ```tprd``` and ```tpld``` (http://tfel.sourceforge.net/documents/mfront/finitestrain.pdf) since in my case either ```A``` or ```B``` is equal to the identity.

Any thoughts on how to do this in MFront?

Best,

Jean-Michel

**Comment by thelfer** [comment]
Hi @jeanmichelscherer,
Could you give more insights on when does such products appears in practice. I imagine they come from some kind of special derivative, do they ?
Best,
Thomas

**Reply by jeanmichelscherer** [reply]
Yes exactly.
For instance, for ```Ce = transpose(E)*E```, then ```dCe_dE```, the derivative of ```Ce``` with respect to ```E```, is equal to ```delta_il*transpose(E)_jk + transpose(E)_ik*delta_jl```, where ```delta_ij``` is the Kronecker symbol.
Jean-Michel

**Reply by jeanmichelscherer** [reply]
I might have found what I need.
The example described above is handled by ```t2tost2<N, real>::dCdF``` right?
Thanks.

**Reply by thelfer** [reply]
@jeanmichelscherer Exactly :+1:
That's mostly the reason why I asked you the reason why you need those operations !
Let me know if you still need something.
Thomas
P.S. Seems like I need to document `t2tost2<N, real>::dCdF` here: <https://thelfer.github.io/tfel/web/tensors.html>

**Reply by jeanmichelscherer** [reply]
Would be perfect :+1: Thanks again

**Comment by thelfer** [comment, marked as answer]
Done:

- <https://thelfer.github.io/tfel/web/tensors.html#cauchy-green-tensor-and-derivative>
- <https://thelfer.github.io/tfel/web/tensors.html#left-cauchy-green-tensor-and-derivative>
